src/pages/index.js

```diff
@@ -1,65 +1,73 @@
 import { FormValidator } from "../components/FormValidator";
 import "./index.css";
 import { Section } from "../components/Section.js";
-import { UserInfo } from "../components/UserInfo";
 import { PopupWithForm } from "../components/PopupWithForm.js";
 import Card from "../components/Card";
 import { PopupWithImage } from "../components/PopupWithImage";
+import { Api } from "../components/Api";
+import { UserInfo } from "../components/UserInfo";
+import { confirmationPopup } from "../components/confirmationPopup";
+import { avatarPopup } from "../components/AvatarPopup";
 
 (function displayCards() {
-  // Elements
-  const imageClickHandler = ({ name, link }) => {
-    imagePopup.open({ name, link });
-  };
-  const cardTemplate =
-    document.querySelector("#card-template").content.firstElementChild;
-  const galleryList = document.querySelector("#gallery__list");
-
-  const items = [
-    {
-      name: "Yosemite Valley",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/yosemite.jpg",
+  const api = new Api({
+    baseUrl: "https://around-api.en.tripleten-services.com/v1",
+    headers: {
+      authorization: "9b29ae94-bb6f-470b-bccd-5f1bdf13a16a",
+      "Content-Type": "application/json",
     },
+  });
 
-    {
-      name: "Lake Louise",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lake-louise.jpg",
-    },
+  let initialCardsInfo = NaN;
+
+  api.getInitialcards().then((items) => {
+    // handle the response
+    initialCardsInfo = items.reverse();
+    console.log(initialCardsInfo);
+    const cardsSection = new Section(
+      {
+        items,
+        renderer: (cardData) => {
+          console.log("---- oo ----", cardData._id);
+          addCard(cardData.name, cardData.link, cardTemplate, cardData._id);
+        },
+      },
+      galleryList
+    );
+    cardsSection.renderItems();
+  });
 
-    {
-      name: "Bald Mountains",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/bald-mountains.jpg",
-    },
+  // try to incorporate PopupWithForm {problems with inputvalues}
+  const editAvatarPopup = new avatarPopup("#avatar-modal", chageAvatar);
+  const avatarPhoto = document.querySelector(".profile__avatar");
 
-    {
-      name: "Latemar",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/latemar.jpg",
-    },
+  function chageAvatar(inputValue) {
+    console.log(inputValue);
+    avatarPhoto.src = inputValue;
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change

**Please, correct it everywhere**